.github/workflows/pull-request.yml

```diff
@@ -449,3 +449,139 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
           GITHUB_REPOSITORY: ${{ github.repository }}
+
+  Merge-Conflict-Check:
+    name: Check for Merge Conflicts
+    runs-on: ubuntu-latest
+    if: github.actor != 'dependabot[bot]'
+    needs: [Code-Quality-Checks]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Check Mergeable Status via API
+        run: |  
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          max_retries=3
+          retry_delay=5
+          
+          for ((i=1; i<=max_retries; i++)); do
+            echo "Attempt $i of $max_retries"
+            
+            response=$(curl -s -f -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              "https://api.github.com/repos/${{ github.repository }}/pulls/$PR_NUMBER")
+            
+            if [ $? -ne 0 ]; then
+              echo "Failed to call GitHub API"
+              if [ $i -eq $max_retries ]; then
+                exit 1
+              fi
+              sleep $retry_delay
+              continue
+            fi
+            
+            mergeable=$(echo "$response" | jq -r '.mergeable')
+            if [ "$mergeable" == "true" ]; then
+              echo "No conflicts detected."
+              exit 0
+            elif [ "$mergeable" == "false" ]; then
+              echo "Merge conflicts detected."
+              exit 1
+            else
+              echo "Mergeable status unknown."
+              if [ $i -eq $max_retries ]; then
+                exit 1
+              fi
+              sleep $retry_delay
+            fi
+          done
+  
+  generate-docs:
+    name: Generate Auto Documentation
+    runs-on: ubuntu-latest
+    needs: [Code-Quality-Checks, Test-Application, Start-App-Without-Docker, Docker-Start-Check]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Required for getting full git history
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22.x'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: |
+          npm ci
+          npm install -g typedoc typedoc-plugin-markdown
+
+      - name: Create documentation directory
+        run: |
+          mkdir -p docs/docs/auto-docs
+          mkdir -p docs/docs/user-guide
```

Comment:
You don't have to create a directory here

1. Create:
    1. `docs/docs/auto-docs`, 
    2. `docs/docs/user-guide`, 
    3. `docs/docs/developer-guide/reference`
4. Add an empty .gitignore file in each.
5. They will be created in the repo and added to your PR